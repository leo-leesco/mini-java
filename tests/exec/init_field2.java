class A {
    String x;
    int y;
    boolean z;
    A next;
    A() {
        x = "toto";
        y = 42;
        z = true;
        next = this;
    }
}
class Main {
    public static void main(String args[]) {
        A a = new A();
        System.out.print("x = " + a.x + "\n");
        System.out.print("y = " + a.y + "\n");
        if (a.z)
            System.out.print("z = true\n");
        if (a.next == a)
            System.out.print("next = a\n");
    }
}

