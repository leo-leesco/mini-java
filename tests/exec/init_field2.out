x = toto
y = 42
z = true
next = a
